src/Handler/AddPrefixCommandHandler.php

```diff
@@ -58,43 +58,65 @@ public function __construct()
      *
      * @return int Returns 0 on success and a positive integer on error.
      */
-    public function handle(Args $args, IO $io)
+    public function handle($prefix, array $paths, BasicFormatter $formatter)
     {
-        $prefix = rtrim($args->getArgument('prefix'), '\\');
-        $paths = $args->getArgument('path');
+        $prefix = rtrim($prefix, '\\');
+        $pathsToSearch = [];
+        $filesToAppend = [];
 
         foreach ($paths as $path) {
             if (!$this->filesystem->isAbsolutePath($path)) {
                 $path = getcwd().DIRECTORY_SEPARATOR.$path;
             }
 
-            if (is_dir($path)) {
-                $this->finder->files()->name('*.php')->in($path)->sortByName();
-
-                foreach ($this->finder as $file) {
-                    $this->scopeFile($file->getPathName(), $prefix, $io);
-                }
+            if (($exists = !file_exists($path)) || !is_readable($path)) {
+                $issue = $exists ? 'does not exist' : 'is not readable';
+                throw new RuntimeException(sprintf(
+                    'A given path %s: %s',
+                    $issue,
+                    $path
+                ));
             }
 
-            if (!is_file($path)) {
-                continue;
+            if (is_dir($path)) {
+                $pathsToSearch[] = $path;
+            } else {
+                $filesToAppend[] = $path;
             }
-
-            $this->scopeFile($path, $prefix, $io);
         }
 
+        $this->finder->files()
+            ->name('*.php')
+            ->in($pathsToSearch)
+            ->append($filesToAppend)
+            ->sortByName();
+
+        $this->scopeFiles($prefix, $formatter);
+
         return 0;
     }
 
-    private function scopeFile($path, $prefix, IO $io)
+    private function scopeFiles($prefix, BasicFormatter $formatter)
+    {
+        $count = count($this->finder);
+        $formatter->outputFileCount($count);
+
+        if ($count > 0) {
```

Comment:
I don't think the count is necessary, the foreach will simply not loop if the count is `0`